src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/AppointmentService.kt

```diff
@@ -14,54 +16,67 @@ import javax.transaction.Transactional
 @Transactional
 class AppointmentService(
   val appointmentRepository: AppointmentRepository,
+  val communityAPIBookingService: CommunityAPIBookingService,
   val authUserRepository: AuthUserRepository,
 ) {
   fun createOrUpdateAppointment(
+    referral: Referral,
     appointment: Appointment?,
     durationInMinutes: Int,
     appointmentTime: OffsetDateTime,
+    appointmentType: AppointmentType,
     createdByUser: AuthUser
   ): Appointment {
-    val noAppointment = appointment == null
-    if (noAppointment) {
-      return createAppointment(durationInMinutes, appointmentTime, createdByUser)
+
+    val initialAppointmentRequired = appointment == null
+    if (initialAppointmentRequired) {
+      val deliusAppointmentId = communityAPIBookingService.book(referral, null, appointmentTime, durationInMinutes, appointmentType)
+      val createdAppointment = createAppointment(durationInMinutes, appointmentTime, deliusAppointmentId, createdByUser)
+      return appointmentRepository.save(createdAppointment)
     }
 
-    val appointmentWithoutAttendanceRecorded = appointment!!.attended == null
-    if (appointmentWithoutAttendanceRecorded) {
-      return updateAppointment(appointment, durationInMinutes, appointmentTime)
+    val updateCurrentAppointmentRequired = appointment!!.attended == null
+    if (updateCurrentAppointmentRequired) {
+      val deliusAppointmentId = communityAPIBookingService.book(referral, appointment, appointmentTime, durationInMinutes, appointmentType)
+      val updatedAppointment = updateAppointment(appointment, durationInMinutes, appointmentTime, deliusAppointmentId)
+      return appointmentRepository.save(updatedAppointment)
     }
 
-    val appointmentWithNonAttendance = appointment.attended == Attended.NO
-    if (appointmentWithNonAttendance) {
-      return createAppointment(durationInMinutes, appointmentTime, createdByUser)
+    val additionalAppointmentRequired = appointment.attended == Attended.NO
+    if (additionalAppointmentRequired) {
```

Comment:
_observation_ if all `createOrUpdateAppointment` calls would use `createAppointment` we essentially wouldn't need 3 branches, just

```kt
val deliusAppointmentId = communityAPIBookingService.book(referral, appointment, appointmentTime, durationInMinutes, appointmentType)
val appointment = createAppointment(durationInMinutes, appointmentTime, deliusAppointmentId, createdByUser)
return appointmentRepository.save(appointment)
```

the dangling appointments are negligible; it would however give us the history of the potentially different users who create/move them around before attendance

Reply:
I think that was the intent of this structure